examples/nidigital_spi/measurement.py

```diff
@@ -0,0 +1,184 @@
+@measurement_service.configuration(
+    "pin_names",
+    nims.DataType.PinArray1D,
+    ["SPI_PINS"],
+    instrument_type=nims.session_management.INSTRUMENT_TYPE_NI_DIGITAL_PATTERN,
+)
+@measurement_service.configuration(
+    "specification_file_path", nims.DataType.Path, "Specifications.specs"
+)
+@measurement_service.configuration("levels_file_path", nims.DataType.Path, "PinLevels.digilevels")
+@measurement_service.configuration("timing_file_path", nims.DataType.Path, "Timing.digitiming")
+@measurement_service.configuration("pattern_file_path", nims.DataType.Path, "Pattern.digipat")
+# TODO: MeasurementLink UI Editor doesn't support arrays of Booleans
+@measurement_service.output("passing_sites", nims.DataType.Int32Array1D)
+@measurement_service.output("failing_sites", nims.DataType.Int32Array1D)
+def measure(
+    pin_names: Iterable[str],
+    specifications_file_path: str,
+    levels_file_path: str,
+    timing_file_path: str,
+    pattern_file_path: str,
+) -> Tuple:
+    """Test a SPI device using an NI Digital Pattern instrument."""
+    logging.info("Starting test: pin_names=%s", pin_names)
+
+    session_management_client = nims.session_management.Client(
+        grpc_channel=measurement_service.get_channel(
+            provided_interface=nims.session_management.GRPC_SERVICE_INTERFACE_NAME,
+            service_class=nims.session_management.GRPC_SERVICE_CLASS,
+        )
+    )
+
+    with contextlib.ExitStack() as stack:
+        reservation = stack.enter_context(
+            session_management_client.reserve_sessions(
+                context=measurement_service.context.pin_map_context,
+                pin_or_relay_names=pin_names,
+                instrument_type_id=nims.session_management.INSTRUMENT_TYPE_NI_DIGITAL_PATTERN,
+                # If another measurement is using the session, wait for it to complete.
+                # Specify a timeout to aid in debugging missed unreserve calls.
+                # Long measurements may require a longer timeout.
+                timeout=60,
+            )
+        )
+
+        if len(reservation.session_info) != 1:
+            measurement_service.context.abort(
+                grpc.StatusCode.INVALID_ARGUMENT,
+                f"Unsupported number of sessions: {len(reservation.session_info)}",
+            )
+
+        session_info = reservation.session_info[0]
+        session = stack.enter_context(_create_nidigital_session(session_info))
+
+        pin_map_context = measurement_service.context.pin_map_context
+        selected_sites_string = ",".join(f"site{i}" for i in pin_map_context.sites)
+        selected_sites = session.sites[selected_sites_string]
+
+        if not session_info.session_exists:
+            # When running the measurement from TestStand, teststand_fixture.py should have
+            # already loaded the pin map, specifications, levels, timing, and patterns.
+            session.load_pin_map(pin_map_context.pin_map_id)
```

Comment:
This example currently assumes that `pin_map_id` is the absolute path. `teststand_fixture.py` has the same assumption: https://github.com/ni/measurementlink-python/blob/main/examples/nidmm_measurement/_helpers.py#L51

Reply:
Fixed: `pin_map_id = teststand_support.get_active_pin_map_id()`